### PR TITLE
ZCS-916:Junit tests for ModifyFilterRules for ElseRules test

### PR DESCRIPTION
**Pull request by sneha-patil-synacor**
Added Junit tests in ModifyFilterRules request for ElseRules test

### CODE REVIEW

**Comment by shrikant-prasad** [comment]
should add two more tests for negative scenarios: elsif after else : if() else{} elsif(){}
 and else nested in another else : if(){} else { else {} }

**Comment by sneha-patil-synacor** [comment]
Added negative test case screnarios

File: store/src/java-test/com/zimbra/cs/filter/ModifyFilterRulesForElseRulesTest.java [hunk 1]
```diff
@@ -0,0 +1,496 @@
...
+    Element filterTests = rule.addElement(MailConstants.E_FILTER_TESTS);
+    filterTests.addAttribute(MailConstants.A_CONDITION, "anyof");
+    Element headerTest = filterTests.addElement(MailConstants.E_HEADER_TEST);
+    headerTest.addAttribute(MailConstants.A_HEADER, "Subject");
+    headerTest.addAttribute(MailConstants.A_STRING_COMPARISON, "is");
+    headerTest.addAttribute(MailConstants.A_VALUE, "important");
+    headerTest.addAttribute(MailConstants.A_INDEX, "0");
+    Element elseRules = rule.addElement(MailConstants.E_ELSE_RULES);
+    Element elseRule1 = elseRules.addElement(MailConstants.E_ELSE_RULE);
+    Element actionLog1 = elseRule1.addElement(MailConstants.E_ACTION_LOG);
```
Comment (shrikant-prasad) [inline store/src/java-test/com/zimbra/cs/filter/ModifyFilterRulesForElseRulesTest.java]: action log should be added inside filterAction.
